WcaOnRails/app/models/registration.rb

```diff
@@ -102,7 +107,7 @@ def waiting_list_info
 
   validate :must_register_for_gte_one_event
   private def must_register_for_gte_one_event
-    if events.empty?
+    if registration_events.length { |r| !r.marked_for_destruction? } == 0
```

Comment:
I took some time to look at the db structure and the interaction between the relevant models and it works a bit differently than what I imagined in my previous comment.

Given your explanation about nested attributes I think we should always check for the `marked_for_destruction` when we offer to modify a `registration_events` list.
Maybe we could extract this check in a specific method on `registration_events` (such as `no_events_selected?` or something)?